### PR TITLE
develop

### PR DESCRIPTION
**Pull request by dkepamprotonme**

### CODE REVIEW

File: AsyncAwait.Task1.CancellationTokens/Program.cs [hunk 1]
```diff
@@ -17,7 +18,7 @@ internal class Program
     /// The Main method should not be changed at all.
     /// </summary>
     /// <param name="args"></param>
-    private static void Main(string[] args)
+    private static void Main()
```
Comment (Ungerfall) [inline AsyncAwait.Task1.CancellationTokens/Program.cs]: Please make Main async. C# 7.1 introduced this possibility

File: AsyncAwait.Task1.CancellationTokens/Calculator.cs [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+            long sum = 0;
+            for (var i = 0; i < n; i++)
+            {
+                if (token.IsCancellationRequested)
+                {
+                    return -1;
+                }
+                // i + 1 is to allow 2147483647 (Max(Int32)) 
+                sum += (i + 1);
+                Thread.Sleep(10);
```
Comment (Ungerfall) [inline AsyncAwait.Task1.CancellationTokens/Calculator.cs]: Do not mix non-blocking and blocking code - this is the way toward deadlocks. There is an alternative non-blocking `Task.Delay`

**Review by Ungerfall — CHANGES_REQUESTED** [review]

File: AsyncAwait.Task1.CancellationTokens/Program.cs (hunk 1 above)
Comment (dkepamprotonme) [reply AsyncAwait.Task1.CancellationTokens/Program.cs]: fixed

File: AsyncAwait.Task1.CancellationTokens/Calculator.cs (hunk 2 above)
Comment (dkepamprotonme) [reply AsyncAwait.Task1.CancellationTokens/Calculator.cs]: fixed

Comment (Ungerfall) [reply AsyncAwait.Task1.CancellationTokens/Calculator.cs]: @dkepamprotonme `Task.Delay` is awaitable

File: AsyncAwait.Task1.CancellationTokens/Program.cs (hunk 1 above)
Comment (Ungerfall) [reply AsyncAwait.Task1.CancellationTokens/Program.cs]: By making `async` I meant all the nested code should non-blocking. One should not mix blocking and non-blocking code

**Pull request merged**